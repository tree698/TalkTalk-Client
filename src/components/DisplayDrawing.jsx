import React from 'react';
import { useNavigate } from 'react-router-dom';
import LoggedInUser from './LoggedInUser';

export default function DisplayDrawing({
  drawing,
  checkboxForDelete,
  toBeDeletedId,
}) {
  const { id, fileName, title, username, brush, description, photo } = drawing;
  const navigate = useNavigate();

  const STYLE_FOR_LOGGEDINUSER = 'flex items-center justify-center shrink-0';
  const STYLE_FOR_AVATAR =
    'w-4 md:w-5 lg:w-6 h-4 md:h-5 lg:h-6 rounded-full mr-0 md:mr-1 lg:mr-2';
  const STYLE_FOR_USERNAME = 'hidden md:block text-xs md:text-sm';

  return (
    <section className="flex flex-col">
      {checkboxForDelete && (
        <input
          name="delete"
          id="delete"
          onChange={() => toBeDeletedId(id)}
          type="checkbox"
          data-testid="delete"
          className="w-3 h-3 lg:w-4 lg:h-4 ml-2 mb-1 border border-superLightGray"
        />
      )}
      <li
        onClick={() =>
          navigate('/talk', {
            state: { id, fileName, title, username, brush, description, photo },
          })
        }
        className="border border-superLightGray list-none rounded-xl shadow-lg cursor-pointer overflow-hidden hover:-translate-y-1 hover:brightness-125 transition-all delay-150 duration-300 ease-in-out"
      >
        <img
          src={fileName}
          alt={title}
          className="w-full h-[246px] md:h-[185px] lg:h-[158px]"
        />
        <div className="flex items-center justify-center md:flex-col gap-x-2 md:gap-x-0 p-1 md:p-2 lg:p-3">
          <p className="mb-1 md:mb-2 text-xs md:text-sm lg:text-base font-semibold">
            {title}
          </p>
          <LoggedInUser
            photo={photo}
            username={username}
            styleForAvatar={STYLE_FOR_AVATAR}
            styleForUsername={STYLE_FOR_USERNAME}
            styleForLoggedInUser={STYLE_FOR_LOGGEDINUSER}
          />
        </div>
      </li>
    </section>
  );
}
