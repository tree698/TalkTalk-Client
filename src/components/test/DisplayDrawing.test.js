import { render, screen, waitFor } from '@testing-library/react';
import { Route, useLocation } from 'react-router-dom';
import { withRouter } from '../../test/utils';
import userEvent from '@testing-library/user-event';
import DisplayDrawing from '../DisplayDrawing';
import { fakeDrawing as drawing } from '../../test/drawingData';
import renderer from 'react-test-renderer';

describe('DisplayDrawing', () => {
  const { id, fileName, title, username, brush, description, photo } = drawing;

  it('renders correctly without checkboxForDelete', () => {
    const component = renderer.create(
      withRouter(
        <Route
          path="/"
          element={
            <DisplayDrawing checkboxForDelete={false} drawing={drawing} />
          }
        />
      )
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with checkboxForDelete', () => {
    const component = renderer.create(
      withRouter(
        <Route
          path="/"
          element={
            <DisplayDrawing checkboxForDelete={true} drawing={drawing} />
          }
        />
      )
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('navigate to talk page with drawing info when clicking drawingCard', async () => {
    function LocationStateDisplay() {
      return <pre>{JSON.stringify(useLocation().state)}</pre>;
    }
    render(
      withRouter(
        <>
          <Route path="/" element={<DisplayDrawing drawing={drawing} />} />
          <Route path={`/talk`} element={<LocationStateDisplay />} />
        </>
      )
    );

    const card = screen.getByRole('listitem');
    userEvent.click(card);

    await waitFor(() => {
      expect(
        screen.getByText(
          JSON.stringify({
            id,
            fileName,
            title,
            username,
            brush,
            description,
            photo,
          })
        )
      ).toBeInTheDocument();
    });
  });

  it('send toBedeleteId props when checked the checkbox', async () => {
    const mockOnChecked = jest.fn();
    render(
      withRouter(
        <Route
          path="/"
          element={
            <DisplayDrawing
              drawing={drawing}
              checkboxForDelete={true}
              toBeDeletedId={mockOnChecked}
            />
          }
        />
      )
    );

    const checkbox = screen.getByTestId('delete');

    userEvent.click(checkbox);

    await waitFor(() => {
      expect(checkbox).toBeChecked();
    });
    expect(mockOnChecked).toHaveBeenCalledTimes(1);
  });
});
