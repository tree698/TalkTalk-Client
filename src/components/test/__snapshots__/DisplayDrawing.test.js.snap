// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayDrawing renders correctly with checkboxForDelete 1`] = `
<section
  className="flex flex-col"
>
  <input
    className="w-3 h-3 lg:w-4 lg:h-4 ml-2 mb-1 border border-superLightGray"
    data-testid="delete"
    id="delete"
    name="delete"
    onChange={[Function]}
    type="checkbox"
  />
  <li
    className="border border-superLightGray list-none rounded-xl shadow-lg cursor-pointer overflow-hidden hover:-translate-y-1 hover:brightness-125 transition-all delay-150 duration-300 ease-in-out"
    onClick={[Function]}
  >
    <img
      alt="test"
      className="w-full h-[246px] md:h-[185px] lg:h-[158px]"
      src="https://res.cloudinary.com"
    />
    <div
      className="flex items-center justify-center md:flex-col gap-x-2 md:gap-x-0 p-1 md:p-2 lg:p-3"
    >
      <p
        className="mb-1 md:mb-2 text-xs md:text-sm lg:text-base font-semibold"
      >
        test
      </p>
      <div
        className="flex items-center justify-center shrink-0"
      >
        <div>
          <img
            alt="test"
            className="w-4 md:w-5 lg:w-6 h-4 md:h-5 lg:h-6 rounded-full mr-0 md:mr-1 lg:mr-2"
            src="https://res.cloudinary.com"
          />
        </div>
        <span
          className="hidden md:block text-xs md:text-sm"
        >
          test
        </span>
      </div>
    </div>
  </li>
</section>
`;

exports[`DisplayDrawing renders correctly without checkboxForDelete 1`] = `
<section
  className="flex flex-col"
>
  <li
    className="border border-superLightGray list-none rounded-xl shadow-lg cursor-pointer overflow-hidden hover:-translate-y-1 hover:brightness-125 transition-all delay-150 duration-300 ease-in-out"
    onClick={[Function]}
  >
    <img
      alt="test"
      className="w-full h-[246px] md:h-[185px] lg:h-[158px]"
      src="https://res.cloudinary.com"
    />
    <div
      className="flex items-center justify-center md:flex-col gap-x-2 md:gap-x-0 p-1 md:p-2 lg:p-3"
    >
      <p
        className="mb-1 md:mb-2 text-xs md:text-sm lg:text-base font-semibold"
      >
        test
      </p>
      <div
        className="flex items-center justify-center shrink-0"
      >
        <div>
          <img
            alt="test"
            className="w-4 md:w-5 lg:w-6 h-4 md:h-5 lg:h-6 rounded-full mr-0 md:mr-1 lg:mr-2"
            src="https://res.cloudinary.com"
          />
        </div>
        <span
          className="hidden md:block text-xs md:text-sm"
        >
          test
        </span>
      </div>
    </div>
  </li>
</section>
`;
